spiped/main.c

```diff
@@ -341,7 +344,8 @@ main(int argc, char * argv[])
 	free(K);
 
 	/* Free arrays of resolved addresses. */
-	sock_addr_freelist(sas_t);
+	if (sas_t != NULL)
```

Comment:
I just noticed that `sock_addr_freelist()` does nothing if the pointer is already NULL, so this line isn't strictly necessary.